agent/consul/fsm/snapshot_oss.go

```diff
@@ -13,98 +13,70 @@ import (
-	registerRestorer(structs.IntentionRequestType, restoreLegacyIntention)
-	registerRestorer(structs.ConnectCARequestType, restoreConnectCA)
-	registerRestorer(structs.ConnectCAProviderStateType, restoreConnectCAProviderState)
-	registerRestorer(structs.ConnectCAConfigType, restoreConnectCAConfig)
-	registerRestorer(structs.IndexRequestType, restoreIndex)
-	registerRestorer(structs.ACLTokenSetRequestType, restoreToken)
-	registerRestorer(structs.ACLPolicySetRequestType, restorePolicy)
-	registerRestorer(structs.ConfigEntryRequestType, restoreConfigEntry)
-	registerRestorer(structs.ACLRoleSetRequestType, restoreRole)
-	registerRestorer(structs.ACLBindingRuleSetRequestType, restoreBindingRule)
-	registerRestorer(structs.ACLAuthMethodSetRequestType, restoreAuthMethod)
-	registerRestorer(structs.FederationStateRequestType, restoreFederationState)
-	registerRestorer(structs.SystemMetadataRequestType, restoreSystemMetadata)
-	registerRestorer(structs.ServiceVirtualIPRequestType, restoreServiceVirtualIP)
-	registerRestorer(structs.FreeVirtualIPRequestType, restoreFreeVirtualIP)
-	registerRestorer(structs.PeeringWriteType, restorePeering)
-	registerRestorer(structs.PeeringTrustBundleWriteType, restorePeeringTrustBundle)
-	registerRestorer(structs.PeeringSecretsWriteType, restorePeeringSecrets)
-}
-
-func persistOSS(s *snapshot, sink raft.SnapshotSink, encoder *codec.Encoder) error {
-	if err := s.persistVirtualIPs(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistNodes(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistSessions(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistACLs(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistKVs(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistTombstones(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistPreparedQueries(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistAutopilot(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistLegacyIntentions(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistConnectCA(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistConnectCAProviderState(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistConnectCAConfig(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistConfigEntries(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistFederationStates(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistSystemMetadata(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistIndex(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistPeerings(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistPeeringTrustBundles(sink, encoder); err != nil {
-		return err
-	}
-	if err := s.persistPeeringSecrets(sink, encoder); err != nil {
-		return err
-	}
-	return nil
-}
+func RegisteredRestorer() map[structs.MessageType]restorer {
```

Comment:
same comment as `commands_oss.go`. Could the common code go to `snapshot.go`, leaving only `registerEnterpriseRestorers` in the oss-version?

Reply:
do you need ``registerEnterprisePersisters`` or is it only for ``Restorers``.

Reply:
Also should I add 
```
//go:build !consulent
```
to the top of ``snapshot_oss.go`` as well or is it just for ``commands_oss.go``?

Reply:
Yeah please add it! We would want `registerEnterprisePersisters` as well.